src/mqueryfront/src/QueryMonaco.js

```diff
@@ -0,0 +1,163 @@
+                monaco.languages.registerCompletionItemProvider("yara", {
+                    provideCompletionItems: () => {
+                        var suggestions = [
+                            {
+                                label: "rule",
+                                kind:
+                                    monaco.languages.CompletionItemKind.Snippet,
+                                insertText: YARA.COMPLETION_RULE,
+                                insertTextRules:
+                                    monaco.languages
+                                        .CompletionItemInsertTextRule
+                                        .InsertAsSnippet,
+                                documentation: "Generate a rule skeleton",
+                            },
+                        ];
+                        return { suggestions: suggestions };
+                    },
+                });
+
+                // We want to have a way to indicate the user that the
+                // Editor object is currently in read-only mode
+                monaco.editor.defineTheme("readOnlyTheme", {
+                    base: "vs",
+                    inherit: true,
+
+                    rules: [
+                        {
+                            token: "",
+                            foreground: "565656",
+                            background: "ededed",
+                        },
+                    ],
+                    colors: {
+                        "editor.background": "#ededed",
+                        "editor.foreground": "#121212",
+                    },
+                });
+
+                // A dirty hack to clear all decorations when the user
+                // started typing after the error has showed
+                editor.onDidChangeModelContent((ev) => {
+                    if (this.state.decorations) {
+                        editor.deltaDecorations(this.state.decorations, [
+                            {
+                                range: new monaco.Range(1, 1, 1, 1),
+                                options: {},
+                            },
+                        ]);
+                    }
+                });
+            })
+            .catch((error) =>
+                console.error(
+                    "An error occurred during initialization of Monaco: ",
+                    error
+                )
+            );
+    }
+
+    render() {
+        return (
+            <Editor
+                name="rawYara"
+                height="70vh"
+                language="yara"
+                theme={this.state.readOnly ? "readOnlyTheme" : "light"}
+                value={this.state.rawYara}
+                editorDidMount={this.handleEditorDidMount}
+                options={{
+                    selectOnLineNumber: true,
+                    lineNumbersMinChars: 0,
+                    glyphMargin: true,
+                    readOnly: this.state.readOnly,
```

Comment:
I think that readOnly should be passed directly from the props and not to be part of state. Avoid using methods like `setReadonly`